sentry_sdk/integrations/litestar.py

```diff
@@ -87,6 +87,10 @@ def __init__(self, app, span_origin=LitestarIntegration.origin):
             span_origin=span_origin,
         )
 
+    def _capture_request_exception(self, exc):
+        # type: (Exception) -> None
+        """Ignore exceptions for requests here: we catch them in Litestar.after_exception handler."""
```

Comment:
I am slightly unsure why this function is needed/how it relates to what you are trying to explain in this PR. Could you please explain in a bit more detail?

Also, I think you need a `pass` here:

```suggestion
    def _capture_request_exception(self, exc):
        # type: (Exception) -> None
        """Ignore exceptions for requests here: we catch them in Litestar.after_exception handler."""
        pass
```

Reply:
Currently, Litestar integration catches exceptions raised in middleware twice:

1. In the custom ASGI handler.
2. In the `Litestar.after_exception` handler, which is added in `patch_app_init`.

The custom ASGI handler doesn't filter exceptions using `failed_request_status_codes`, so all exceptions (including those not filtered by the expected status code) appear in Sentry.

Interestingly, this behavior doesn't apply to exceptions raised in request handlers because Litestar prepends the built-in exception handling middleware.

---

This leads us to the conclusion that the ASGI handler only catches exceptions raised in middleware or lifespan handlers. It's still valuable to catch exceptions from lifespan handlers.

In this case, we patch `_capture_request_exception` to prevent catching exceptions raised in middleware.